src/main.py

```diff
@@ -11,17 +11,69 @@
 
 def welcome(update, context):
     for new_user_obj in update.message.new_chat_members:
-        user = str(new_user_obj['first_name']) + " " + str(new_user_obj['last_name'])
+        if( new_user_obj['last_name'] is None):
+            user = str(new_user_obj['first_name'])
+        else:
+            user ="(" + str(new_user_obj['first_name']) + " " + str(new_user_obj['last_name']) + ")"
+
+        if( new_user_obj['username'] is None):
+            username = ""
+        else:
+            username = "@" + str(new_user_obj['username']) + " "
         context.bot.send_message(chat_id=update.message.chat_id,
-                                 text='Halo ' + str(user) + ' ,'
+                                 text='Halo '+ str(username) + str(user) + ' ,'
                                                             '\n\nSelamat datang di grup Attack On IF! '
                                                             '\nSalam kenal yaaa!')
+def echo(update, context):
```

Comment:
The function name is not clear and too generic. You can rename it to `send_welcome_message`